### DISCUSSION TITLE
HOGWARTS PROJECT: Dynamic form not showing on DOM on button click

### DISCUSSION THREAD

**Discussion by cmeffley** [nss-evening-cohort-14/evening-client-side#60] [Help Me!] [answered]
Chris Meffley

I can't get the form to show up in the DOM, I don't think I'm making the correct function.

No other help tickets

Error messages for the loop function, but have sense changed approach.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click
https://www.w3schools.com/jsref/event_onclick.asp
https://www.w3schools.com/js/js_htmldom_events.asp

I've tried a loop, but realized I don't need that since I want it to run one time. And I'm trying to create another function for the form to populate in the DOM, but I've hit a wall on possibly solutions.

**Comment by TrinityChristiana** [comment]
Looks like the google results you sent might be difficult to read because they are adding the functionality directly in the HTML. Here are how we did it in class:
- Here's an example of how we added items to the dom in the event-listeners walk though: [Printing to dom example](https://github.com/nss-evening-cohort-14/event-listeners/blob/create-pies/main.js#L66)
- Here's an example of how to make something happen on a button click: [On click callback](https://github.com/nss-evening-cohort-14/event-listeners/blob/create-pies/main.js#L162)

Also, this is how to toggle it using styling: [How TO - Toggle Hide and Show
](https://www.w3schools.com/howto/howto_js_toggle_hide_show.asp) 
- The example shows you how to add the on click directly to the HTML, but we want to do that in the JS file instead, using a query selector.

If you need any more assistance, can you attach a code snippet of what you do have? Thank you!

**Comment by cmeffley** [comment]
This is where I'm currently at after trying some other things (for loop, just to see if it will work, among other things). I'm confused on what function/how to put it in a function to get that form to show up:
```
const createForm = (taco) => {
  let domString = '';

  `<form>
    <div class="mb-3">
    <label for="student" class="form-label">Student</label>
    <input type="text" class="form-control" id="student" required>
    </div>
    <button type="submit" class="btn btn-primary">Sort!</button>
  </form>`
  };
```

Sorry, hard to move on, my mind wants to figure this out!

**Reply by cmeffley** [reply]
![2021-01-30](https://user-images.githubusercontent.com/76795299/106366394-91636d80-6309-11eb-9a88-fb0363c044c7.png)

This should  be the missing code

**Reply by TrinityChristiana** [reply]
Oh, I see now! You do not need to loop over anything to create the `domString`.

**Reply by drteresavasquez** [reply]
One thing you can do is put your form directly on the DOM like we did in pies to have the form and then refactor after.

**Comment by drteresavasquez** [comment]
@cmeffley if you have resolved this let us know. This is very similar to #61 which seems to be resolved.

**Reply by cmeffley** [reply]
I was in that general area, but that definitely helped!

**Reply by drteresavasquez** [reply]
@cmeffley can you please finish out this ticket by telling us what you did to solve your problem in a new comment?

**Comment by cmeffley** [comment, marked as answer]
I realized since I did not need a loop in order to create the domString, I just needed a function and use the printToDom function to get it on the page. I simplified what I had been trying to do:

```
const createForm = () => {
    let form = `<form>
    <div class="mb-3">
    <label for="student" class="form-label">Student</label>
    <input type="text" class="form-control" id="student" required>
    </div>
    <button type="submit" class="btn btn-info" id="sort1">Sort!</button>
    </form>`
    printToDom('#form', form);
};
```

**Reply by drteresavasquez** [reply]
NOICE!

**Comment by Gabrielle-Tobermann** [comment]
I am having the same issue. My form will not show up when the 'Let's start sorting' button is clicked. I added an empty div in html with an id to target in my buttonEvents functions, it looks right to me but I am clearly missing something...

![sortingHatJS](https://user-images.githubusercontent.com/76187279/106538855-cad6dd00-64b1-11eb-93a1-fc146a4cbca7.png)

![sortingHatHTML](https://user-images.githubusercontent.com/76187279/106538846-c6122900-64b1-11eb-986f-d2b2ef049af2.png)

**Reply by drteresavasquez** [reply]
@Gabrielle-Tobermann push your code up and give us a link to the file if possible. It makes it easier for us to give specific details on the code.

**Reply by Gabrielle-Tobermann** [reply]
There was no actual issue with my code, I had created two init functions by accident!

**Reply by drteresavasquez** [reply]
That is good that you found the issue! Bravo!
